### ISSUE TITLE
["error", {}] breaks option defaults (no-construct-stack-suffix disabled; require-passing-this flips to strict)

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#538]
### Describe the bug

Configuring a rule as `["error", {}]` (an empty options object — a natural way to write "use the defaults") produces behavior different from `["error"]`, because the rules read `context.options[0] ?? defaultOption` / `|| defaultOption`: an empty object is truthy, so the hand-rolled default is skipped and the rule runs with an option object whose keys are missing (or filled from a JSON-schema `default` that contradicts the documented default). Two rules are affected:

**1. `no-construct-stack-suffix` — rule silently disabled.** With `{}`, `options.disallowedSuffixes` is `undefined`, both `disallowedSuffixes?.includes(...)` checks short-circuit, and nothing is ever reported.

```ts
// linted with "awscdk/no-construct-stack-suffix": ["error", {}]
export class SuffixEmptyOption extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);
    new Bucket(this, "SampleBucketConstruct"); // reported with ["error"], NOT reported with ["error", {}]
  }
}
```

**2. `require-passing-this` — default flips to strict mode.** The schema declares `allowNonThisAndDisallowScope: { default: false }` while the documented and coded default is `true`. AJV (`useDefaults: true`, in both ESLint and oxlint) rewrites `{}` to `{ allowNonThisAndDisallowScope: false }`, so `["error", {}]` behaves as strict mode while `["error"]` behaves as allow mode.

```ts
export class OptionSample extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);
    const sample = new Bucket(this, "Sample");
    new Bucket(sample, "Child"); // allowed with ["error"], reported with ["error", {}]
  }
}
```

### Expected Behavior

`["error", {}]` behaves exactly like `["error"]`:

- `no-construct-stack-suffix` keeps its default `disallowedSuffixes: ["Construct", "Stack"]`
- `require-passing-this` keeps its documented default `allowNonThisAndDisallowScope: true`

### Current Behavior

- `no-construct-stack-suffix` with `["error", {}]`: exit code 0, no diagnostics at all (both plugins; with `["error"]` the same file reports `Construct ID 'SampleBucketConstruct' should not include Construct suffix`).
- `require-passing-this` with `["error", {}]`:

```
repro/require-passing-this-options/empty/option.ts  9:16  error  Require passing `this` in a constructor  awscdk/require-passing-this
```

while the identical file linted with `["error"]` reports nothing. Oxlint behaves identically in both cases.

### Plugin

eslint-plugin-awscdk, oxlint-plugin-awscdk

### Plugin Version

eslint-plugin-awscdk 4.4.0, oxlint-plugin-awscdk 1.0.0

### Language Version

TypeScript 6.0.3 (eslint-plugin-awscdk) / TypeScript 7 preview via oxlint-tsgolint (oxlint-plugin-awscdk)

### Additional Context

- Verified on `main` (35b889a), macOS, rule-isolated configs (`["error"]` vs `["error", {}]` on identical files).
- Root causes:
  - `packages/*/src/rules/no-construct-stack-suffix.ts` — `context.options[0] ?? defaultOption`; the top-level `defaultOptions` passed to the rule creator is not merged into `context.options` (typescript-eslint's RuleCreator only passes merged options as the unused second `create` argument; oxlint only merges `meta.defaultOptions`)
  - `packages/*/src/rules/require-passing-this.ts` — schema `default: false` contradicts `defaultOption` / docs (`allowNonThisAndDisallowScope: true`); AJV `useDefaults` injects the schema value into `{}`
- Possible fixes: merge partial option objects over the rule defaults (spread), and align the schema `default` with the documented default (or drop it).
